litestar/contrib/sqlalchemy/repository.py

```diff
@@ -59,23 +61,23 @@ def wrap_sqlalchemy_exception() -> Any:
         raise RepositoryError(f"An exception occurred: {exc}") from exc
 
 
-class SQLAlchemyRepository(AbstractRepository[ModelT], Generic[ModelT]):
+class SQLAlchemyAsyncRepository(AbstractAsyncRepository[ModelT], Generic[ModelT]):
     """SQLAlchemy based implementation of the repository interface."""
 
-    def __init__(
-        self, *, session: AsyncSession, base_select: Select[tuple[ModelT]] | None = None, **kwargs: Any
-    ) -> None:
+    match_fields: list[str] | str | None = None
+
+    def __init__(self, *, statement: Select[tuple[ModelT]] | None = None, session: AsyncSession, **kwargs: Any) -> None:
         """Repository pattern for SQLAlchemy models.
 
         Args:
+            statement: To facilitate customization of the underlying select query.
             session: Session managing the unit-of-work for the operation.
-            base_select: To facilitate customization of the underlying select query.
             **kwargs: Additional arguments.
 
         """
         super().__init__(**kwargs)
         self.session = session
-        self.statement = base_select if base_select is not None else select(self.model_type)
+        self.statement = statement if statement is not None else select(self.model_type)
```

Comment:
```suggestion
        self.statement = statement or select(self.model_type)
```

I don't know if there's a condition under which a `Select` could ever be considered falsey. If there is, my suggestion won't work :grimacing:

Reply:
I had it this way before but ran into a few errors, i think.  I'll change it back and do some testing.